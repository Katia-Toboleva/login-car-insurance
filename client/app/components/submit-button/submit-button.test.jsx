import React from 'react';
import { shallow } from 'enzyme';
import SubmitButton from './submit-button';

const getUnit = (props) => (
  shallow(<SubmitButton {...props} />)
);

describe('SubmitButton', () => {
  describe('render', () => {
    it('should render correctly', () => {
      const wrapper = getUnit();

      expect(wrapper).toMatchSnapshot();
    });

    describe('handleMouseEnter', () => {
      it('updates html correctly', () => {
        const wrapper = getUnit();
        wrapper.find('.submit-button').simulate('mouseEnter');

        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('handleMouseLeave', () => {
      it('updates html correctly', () => {
        const wrapper = getUnit();
        wrapper.find('.submit-button').simulate('mouseLeave');

        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
