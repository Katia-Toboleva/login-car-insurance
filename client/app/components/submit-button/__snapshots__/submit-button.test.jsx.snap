// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton render handleMouseEnter updates html correctly 1`] = `
<input
  className="submit-button submit-button--hovered submit-button--disabled"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  type="submit"
  value="sign in"
/>
`;

exports[`SubmitButton render handleMouseLeave updates html correctly 1`] = `
<input
  className="submit-button submit-button--disabled"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  type="submit"
  value="sign in"
/>
`;

exports[`SubmitButton render should render correctly 1`] = `
<input
  className="submit-button submit-button--disabled"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  type="submit"
  value="sign in"
/>
`;
